addon/components/transition/child.js

```diff
@@ -9,6 +11,28 @@ export default class TransitionChildComponent extends Component {
 
     assert(PARENT_MISSING_ERROR_MESSAGE, typeof parent !== 'undefined');
   }
+
+  trackChild = modifier((element) => {
+    this.trackDomNode(element);
+
+    return () => {
+      this.untrackDomNode(element);
+    };
+  });
+
+  @action
+  trackDomNode(element) {
+    if (this.args.trackDomNode) {
```

Comment:
stuff like this could be reduced to `this.args.trackDomNode?.(element)`
and then with the functions move into the modifier directly, we cut the code to 1/3rd :partying_face: